shards/shards.go

```diff
@@ -487,6 +564,18 @@ func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q qu
 	shards, q = selectRepoSet(shards, q)
 	tr.LazyPrintf("after selectRepoSet shards:%d %s", len(shards), q)
 
+	switch v := opts.Visibility; v {
+	case "public", "private":
+		q = &query.And{Children: []query.Q{query.RepoFunc(func(name string) bool {
+			_, ok := ss.public[name]
+			return ok == (v == "public")
```

Comment:
minor perf: can you hoist out the check of `v == "public"` outside of the function to avoid doing that every call.